src/mscorlib/shared/System/Globalization/GregorianCalendar.cs

```diff
@@ -520,8 +520,13 @@ public override DateTime ToDateTime(int year, int month, int day, int hour, int
             throw new ArgumentOutOfRangeException(nameof(era), SR.ArgumentOutOfRange_InvalidEraValue);
         }
 
-        internal override Boolean TryToDateTime(int year, int month, int day, int hour, int minute, int second, int millisecond, int era, out DateTime result)
+        internal override bool TryToDateTime(int year, int month, int day, int hour, int minute, int second, int millisecond, int era, out DateTime result)
         {
+            if (!IsValidDay(year, month, day, era))
+            {
+                result = DateTime.Now;
+                return false;
+            }
             if (era == CurrentEra || era == ADEra)
```

Comment:
> if (era == CurrentEra || era == ADEra) [](start = 11, length = 39)

I think we don't need to check fir era here again as IsValidDay already doing this check

Reply:
Agreed, that check can be removed